Fixes 32548 - Moving salt parser and importer to core

domitea opened the pull request:
Like Ansible parser (https://github.com/theforeman/foreman/pull/8491) and rhsm parser (https://github.com/theforeman/foreman/pull/8508) this PR moving Salt parser from foreman_salt plugin into core.

This PR relies on the Ansible parser PR, becuase it's using initializer made during Ansible PR. 

<!---

Thank you for contributing to The Foreman, please read the
[following guide](https://www.theforeman.org/contribute.html), in short:

* [Create an issue](https://projects.theforeman.org/projects/foreman/issues)
* Reference the issue via `Fixes #1234` in the commit message
* Prefer present-tense, imperative-style commit messages
* Mark all strings for translation, see [1]
* Suggest prerequisites for testing and testing scenarios following example above.
* Prepend `[WIP]` for work in progress to prevent bots from triggering actions
* Be patient, we will do our best to take a look as soon as we can
* Explain the purpose of the PR, attach screenshots if applicable
* List all prerequisites for testing (e.g. VMware cluster, two smart proxies...)
* Reviewers often use extensive list of items to check, have a look prior submitting [2]
* Be nice and respectful

1: https://projects.theforeman.org/projects/foreman/wiki/Translating#Translating-for-developers
2: https://github.com/theforeman/foreman/blob/develop/developer_docs/pr_review.asciidoc
-->

domitea committed:
Fixes #32548 - Moving Salt fact parser to Core

Moving Salt fact parser from Salt plugin to Core for better refactoring of parsers logic

lzap commented:
I am missing integration test here too, I hope the `chef` command is able to print its facts in JSON or YAML easily.

domitea commented:
I understand your concern and I know that testing is very important part of developing. I'm planning also refactor tests for parsers and importers together with refactoring of these components. That means I need firstly have all parsers and importers in one place for refactoring.

ares commented:
when the time comes, I can prepare the chef facts example, you could get it from `ohai` however the chef-handler-foreman (the client bit we install to every client) modifies the payload to be more Foreman friendly.

lzap commented:
We are merging Ansible PR soon, unless there are any objections I am going to merge these parser PRs into core so @domitea  can start refactoring them.

sbernhard commented:
What is the state here @domitea ? (Thanks a lot for this PR and the PR regarding this change in foreman_salt). Very appreciated!

ekohl reviewed (changes requested):
Needs a rebase now.

domitea commented:
@sbernhard Well, the Ansible parser is merged to core. The PR for Ansible plugin will be merged soon I hope. So after that I can continue on other PRs (RHSM parser from Katello, Chef and Salt). 

So stay tuned, more changes will come!

ekohl commented:
Here we have the same concern as Chef: are we license wise allowed to include this <del>file</del> the logo in our repository?

domitea commented:
I've updated a PR and tests are green... Conflict is fixed and I changed the Salt icon with ordinary one. So I think It's ready to merge. What do you think @ekohl, @lzap, @ares and @sbernhard ?

sbernhard commented on app/models/fact_names/foreman_salt/fact_name.rb:
```diff
@@ -0,0 +1,14 @@
+    end
+
+    def icon_path
+      'icons16x16/stub/black-s.png'
```
We can use this one for now. But, foreman_salt used the salt original icon since years. I recommend to contact saltproject@vmware.com to ask for permission to use their logo.

tbrisker reviewed (approved):
Thanks @domitea ! LGTM, failures on the salt PR don't seem to be related.

lzap commented:
Some holidays accumulated GH notifications, thanks for the patch!
